src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerPanel.tsx

```diff
@@ -75,11 +69,51 @@ export const TemplatesCatalogDrawerPanel: React.FC<TemplatesCatalogDrawerPanelPr
     const interfaces = getTemplateInterfaces(template);
     const disks = getTemplateDisks(template);
     const isDefaultTemplate = isDefaultVariantTemplate(template);
-    const { memory, cpuCount } = getTemplateFlavorData(template);
     const hostDevicesCount = getHostDevices(vmObject)?.length || 0;
     const gpusCount = getGPUDevices(vmObject)?.length || 0;
     const hardwareDevicesCount = hostDevicesCount + gpusCount;
 
+    const [updatedVMObject, setUpdatedVMObject] = useState<V1VirtualMachine>(undefined);
+    const [error, setError] = useState<any>();
+
+    useEffect(() => {
+      setError(undefined);
+
+      k8sCreate<V1Template>({
+        model: ProcessedTemplatesModel,
+        data: template,
+        queryParams: {
+          dryRun: 'All',
+        },
+      })
+        .then((processedTemplate) => {
+          setUpdatedVMObject(getTemplateVirtualMachineObject(processedTemplate));
+        })
+        .catch((err) => {
+          setError(err);
+        });
+    }, [template]);
+
+    const onCPUMemorySubmit = (vmObj: V1VirtualMachine) => {
+      const updatedVM = produce<V1VirtualMachine>(vmObj, (vmDraft: V1VirtualMachine) => {
+        ensurePath(vmDraft, [
+          'spec.template.spec.domain.resources',
+          'spec.template.spec.domain.cpu',
+        ]);
+
+        vmDraft.metadata.namespace = ns || DEFAULT_NAMESPACE;
+        vmDraft.spec.template.spec.domain.resources.requests = {
+          ...vmDraft?.spec?.template?.spec?.domain?.resources?.requests,
+          memory: `${vmObj.spec.template.spec.domain.resources.requests['memory']}`,
+        };
+        vmDraft.spec.template.spec.domain.cpu.cores = vmObj.spec.template.spec.domain.cpu.cores;
+      });
+
+      setUpdatedVMObject(updatedVM);
```

Comment:
maybe we can extract it to util? just to make the file smaller?
to a func updateVMCPU(VM, callback, ns?) and can be used as  updateVMCPU(vmObj, setUpdatedVMObject, ns)

Reply:
Maybe.. I was thinking about refactoring this code the way that it could be used also in other places where we update cpu and memory values, but I am not sure if it worth, because we wouldn't save much lines of the code anyway.. But maybe the code would be more readable at least..